### PR TITLE
Feature/build 2 ecr

### PR DESCRIPTION
**Pull request by GeorgesLorre**
I :
- created a public repo called `fndnt` ont the correct AWS account (the name reservation is pending)
- altered the `build_components.sh` script. So now It will create repo's if they do not exist yet and add a tag with the public ECR uri. (if you want to run it locally you will need to be logged in in Docker with AWS credentials)
- I ran the script locally for one component and without building to docker hub (only ECR) succesfully
- setup OIDC on the Public AWS account for github actions (using [1](https://www.automat-it.com/post/using-github-actions-with-aws-iam-roles)  [2](https://aws.amazon.com/blogs/security/use-iam-roles-to-connect-github-actions-to-actions-in-aws/))
- updated the github actions to include fetching credentials via OIDC and authenticating to the public ecr. (every AWS account can only have 1 public ECR so all aws commands using `aws ecr-public ...` will be ran against it.

I still need to test everything together in github itself.

### CODE REVIEW

File: scripts/build_components.sh [hunk 1]
```diff
@@ -73,12 +74,17 @@ for dir in "${components_to_build[@]}"; do
...
+    aws_full_image_names+=("$aws_full_image_name")
   done
 
   echo "Updating the image version in the fondant_component.yaml with:"
   echo "${full_image_names[0]}"
   sed -i -e "s|^image: .*|image: ${full_image_names[0]}|" fondant_component.yaml
 
+  # create repo if not exists
+  aws ecr-public describe-repositories --region us-east-1 --repository-names ${BASENAME} || aws ecr-public create-repository --region us-east-1 --repository-name ${BASENAME}
+  full_image_names+=("public.ecr.aws/m4f5a0g0/${BASENAME}:${tag}")
```
Comment (GeorgesLorre) [inline scripts/build_components.sh:86]: This is temporary until we get the `fndnt` name reserved.

**Comment by GeorgesLorre** [comment]
I will need to alter the other github action scripts aswel but want to test this first

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -6,6 +6,14 @@ on:
       - main
   workflow_dispatch:
 
+env:
+
```
Comment (RobbeSneyders) [inline .github/workflows/build.yaml:10]:
```suggestion
```

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -6,6 +6,14 @@ on:
       - main
   workflow_dispatch:
 
+env:
+  
+  AWS_REGION : us-east-1
+
+permissions:
+      id-token: write
+      contents: read
```
Comment (RobbeSneyders) [inline .github/workflows/build.yaml:15]:
```suggestion
  id-token: write
  contents: read
```

File: scripts/build_components.sh (hunk 1 above)
Comment (RobbeSneyders) [reply scripts/build_components.sh:86]: Can you update this now?

**Review by RobbeSneyders — APPROVED** [review]
Thanks @GeorgesLorre!

Feel free to merge after addressing the comments.

**Pull request merged**